src/inject/dynamic-theme/adopted-style-manger.ts

```diff
@@ -0,0 +1,79 @@
+import {Theme} from '../../definitions';
+import {createStyleSheetModifier} from './stylesheet-modifier';
+import {getCSSVariables} from './css-rules';
+
+const adoptedStyleOverrides = new WeakMap<CSSStyleSheet, CSSStyleSheet>();
+const overrideList = new WeakSet<CSSStyleSheet>();
+
+export interface AdoptedStyleSheetManager {
+    render(theme: Theme, variables: Map<string, string>): void;
+    destroy(): void;
+}
+
+export function createAdoptedStyleSheetOverride(node: Document | ShadowRoot): AdoptedStyleSheetManager {
+
+    function injectSheet(sheet: CSSStyleSheet, override: CSSStyleSheet) {
+        const newSheets = [...node.adoptedStyleSheets];
+        const sheetIndex = newSheets.indexOf(sheet);
+        const existingIndex = newSheets.indexOf(override);
+        if (sheetIndex === existingIndex - 1) {
+            return;
+        }
+        if (existingIndex >= 0) {
+            newSheets.splice(existingIndex, 1);
+        }
+        newSheets.splice(sheetIndex + 1, 0, override);
+        node.adoptedStyleSheets = newSheets;
+    }
+
+    function destroy() {
+        const newSheets = [...node.adoptedStyleSheets];
+        node.adoptedStyleSheets.forEach((adoptedStyleSheet) => {
+            if (overrideList.has(adoptedStyleSheet)) {
+                const existingIndex = newSheets.indexOf(adoptedStyleOverrides.get(adoptedStyleSheet));
```

Comment:
Bug, `adoptedStyleSheet` is an override here, `adoptedStyleOverrides.get(adoptedStyleSheet)` is `undefined`